### PR TITLE
Clean up ahead of light-batch work

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

Just some clean-up I wanted to do and TODOs I wanted to add. Also a bug fix in how the full-network sim nodes were created.

### What changes were made as part of this PR:

- Documents the `DB.AddSubmittedRollup` and `DB.WasSubmitted` methods
- Provides a node field to indicate whether we're the sequencer
- Adds some todos
- Fixes bug whereby full-network sim node addresses were empty

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

**Review by tudor-malene — COMMENTED** [review]
super minor check

File: go/host/node/node.go [hunk 1]
```diff
@@ -82,10 +83,15 @@ type Node struct {
 }
 
 func NewHost(config config.HostConfig, stats host.StatsCollector, p2p host.P2P, ethClient ethadapter.EthClient, enclaveClient common.Enclave, ethWallet wallet.Wallet, mgmtContractLib mgmtcontractlib.MgmtContractLib, logger gethlog.Logger) host.MockHost {
+	if config.IsGenesis && config.NodeType != common.Aggregator {
```
[the remark this answers is not in the document]
Comment (joeldudley) [reply go/host/node/node.go:86]: I went for two separate checks, as it allowed me to give clearer error messages.

**Review by tudor-malene — APPROVED** [review]
lgtm

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**